definitions/scenarios/restore.rb

```diff
@@ -6,19 +6,27 @@ class Restore < ForemanMaintain::Scenario
       description 'Restore backup'
       param :backup_dir, 'Path to backup directory'
       param :incremental_backup, 'Is the backup incremental?'
+      param :dry_run, 'Check if backup could be restored, without performing the restore'
       manual_detection
     end
 
     # rubocop:disable Metrics/MethodLength,Metrics/AbcSize
+    # rubocop:disable Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity
     def compose
       backup = ForemanMaintain::Utils::Backup.new(context.get(:backup_dir))
 
       add_steps(find_checks(:root_user))
       supported_version_check
       add_steps_with_context(Checks::Restore::ValidateBackup,
-                             Procedures::Restore::Confirmation,
                              Checks::Restore::ValidateHostname,
-                             Checks::Restore::ValidateInterfaces,
+                             Checks::Restore::ValidateInterfaces)
+
+      if context.get(:dry_run)
+        self.class.metadata[:run_strategy] = :fail_slow
+        return
+      end
```

Comment:
Happy to see a "nicer" implementation of "if dry run, just add checks and fail_slow", but this one works, so :woman_shrugging:

Reply:
It works as expected so good to go from my side.